docs/ASDK-2.0-Layout-Documention.md

```diff
@@ -0,0 +1,802 @@
+  1. the size (but not position) of the layoutable object and
+  2. the size and position of all of its immediate child objects.
+
+`ASDisplayNode` as well as `ASLayoutSpec` classes are conforming to the `ASLayoutable` protocol and provide this functionally out of the box. This allows you to compose nodes and layout specs to build very complex layouts in a declarative way.
+
+### Replaced Layoutables
+A replaced layoutable is any layoutable whose appearance and dimensions are defined by an external resource. Examples include `ASImageNode`, `ASTextNode`, `ASButtonNode` or `ASTextNode`. All other layoutables can be referred to as non-replaced layoutables. This would include `ASVideoNode`, `ASVideoPlayerNode`, `ASNetworkImageNode` or `ASEditableTextNode`.
+
+Replaced layoutables can have intrinsic dimensions—width and height values that are defined by the layoutable itself, rather than by its surroundings. For example, if an `ASImageNode` has a width set to `ASDimensionAuto`, the width of the linked image file will be used. For an `ASImageNode` the intrinsic content size would be calculated based on the text content.
+
+## Height and Width
+
+### The content area
+The content area is the area containing the real content of a layoutable. The properties `width`, `minWidth`, `maxWidth`, `height`, `minHeight` and `maxHeight` control the content size.
+
+### Setting a size
+A layoutable's height and width dimensions determine its size on the screen. The simplest way to set the dimensions of a layoutable is by setting the fixed `width` and `height` property. All dimensions in AsyncDisplayKit's layout system are unitless, and represent density-independent pixels. Setting dimensions this way is common for layoutables that should always render at exactly the same size, regardless of screen dimensions.
+
+```Obj-c
+// Set a height of 0.5 points
+layoutable.width = ASDimensionMake(ASDimensionTypePoints, 0.5);
+
+// Set a width of 50%
+layoutable.width = ASDimensionMake(ASDimensionTypePercentage, 0.5);
+```
+
+Besides width and height there is also `minWidth`, `minHeight`, `maxWidth` and `maxHeight` properties. For example an layoutable to which `minWidth` is applied will never be narrower than the minimum width specified, but it will be allowed to grow normally if its content exceeds the minimum width set. For a full description look at the `Layout Properties` section.
+
+### Sizing Dimensions
+Setting dimensions on layoutables are of the type `ASDimension`. `ASDimension` is essentially a normal CGFloat with support for representing either a point value, or a % value. It allows the same API to take in both fixed values, as well as relative ones:
```

Comment:
The reader should already understand that dimensions are "essentially a normal CGFloat with support for representing either a point value, or a % value." at this point if it's described in the intro. Focus on the fact that this concept is represented by the `ASDimension` type and all the additional utilities that come with it.